src/Features/CSharp/Portable/SplitOrMergeIfStatements/CSharpMergeConsecutiveIfStatementsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,84 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Composition;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+using Microsoft.CodeAnalysis.SplitOrMergeIfStatements;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.SplitOrMergeIfStatements
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = PredefinedCodeRefactoringProviderNames.MergeConsecutiveIfStatements), Shared]
+    internal sealed class CSharpMergeConsecutiveIfStatementsCodeRefactoringProvider
+        : AbstractMergeConsecutiveIfStatementsCodeRefactoringProvider<ExpressionSyntax>
+    {
+        protected override string IfKeywordText => SyntaxFacts.GetText(SyntaxKind.IfKeyword);
+
+        protected override bool IsApplicableSpan(SyntaxNode node, TextSpan span, out SyntaxNode ifStatementNode)
```

Comment:
can we not make thsi more strongly typed?  Even if it would have to be 'SyntaxNode' for VB, it seems like the C# side could be more restrictive.

Reply:
can we not make thsi more strongly typed? Even if it would have to be 'SyntaxNode' for VB, it seems like the C# side could be more restrictive.

Reply:
can we not make thsi more strongly typed? Even if it would have to be 'SyntaxNode' for VB, it seems like the C# side could be more restrictive.

Reply:
can we not make thsi more strongly typed? Even if it would have to be 'SyntaxNode' for VB, it seems like the C# side could be more restrictive.

Reply:
can we not make thsi more strongly typed? Even if it would have to be 'SyntaxNode' for VB, it seems like the C# side could be more restrictive.

Reply:
can we not make thsi more strongly typed? Even if it would have to be 'SyntaxNode' for VB, it seems like the C# side could be more restrictive.